### ISSUE TITLE
Minor bug in do_MLICA and possible bug in apply_ica_replace_mean_std

### ISSUE THREAD

**Issue by pravsripad** [jdammers/jumeg#151]
Hello @ckiefer0 I tried running it on my machine and there seems to be two issues.. 

1. Script fails because bads_corr_ctps is an array and a list is expected. Below change in bold will fix that. 
```
# get ECG/EOG related components using JuMEG
ic_ecg = get_ics_cardiac(raw_filtered_chop, ica, flow=flow_ecg, fhigh=fhigh_ecg,
                         thresh=ecg_thresh, tmin=-0.5, tmax=0.5,
                         name_ecg=ecg_ch, use_CTPS=True)**[0]  # returns both ICs and scores (take only ICs)**
ic_eog = get_ics_ocular(raw_filtered_chop, ica, flow=flow_eog, fhigh=fhigh_eog,
                        thresh=eog_thresh, name_eog_hor=eog1_ch,
                        name_eog_ver=eog2_ch, score_func='pearsonr')

bads_corr_ctps = list(ic_ecg) + list(ic_eog)
```
2. Upon plotting the performance of the artefact removal, something doesn't seem to work correctly, data doesn't seem to be cleaned. This can be tested by including the below lines are the very end of the do_MLICA.py .

```
fnout_fig = '109925_CAU01A_100715_0842_2_c,rfDC,0-45hz,ar-perf'
plot_performance_artifact_rejection(raw_chop, ica, fnout_fig, meg_clean=raw_chop_clean,
                                    name_ecg='ECG 001', name_eog='EOG 002')
```

**Comment by ckiefer0** [comment]
Hello @pravsripad,

I fixed the ic_ecg bug.

Regarding your second statement, no bad components have actually been defined so ica.exclude should be an empty list. I intended the script to just show the difference between the three cases and to caution the user that, depending on what method they use, they will get different results.

**Comment by ckiefer0** [comment]
I added the following lines at the end of the file and the performance plot looked good to me:

    from jumeg.jumeg_plot import plot_performance_artifact_rejection
    fnout_fig = '109925_CAU01A_100715_0842_2_c,rfDC,0-45hz,ar-perf'

    raw_copy = raw.copy().crop(tmin=tmin*4./1000, tmax=tmax*4./1000)

    plot_performance_artifact_rejection(raw_copy, ica, fnout_fig,
                                        meg_clean=raw_chop_clean,
                                        show=True, verbose=False,
                                        name_ecg=ecg_ch,
                                        name_eog=eog2_ch)

![109925_cau01a_100715_0842_2_c rfdc 0-45hz ar-perf](https://user-images.githubusercontent.com/25769508/41457933-7bd67288-7085-11e8-876c-956ec92921a0.png)

**Comment by pravsripad** [comment]
Hi, the above works for me as well, but it is not using the newly included apply_ica_replace_mean_std routine . If we use the raw objects returned by the above routine, then the ica doesn't seem to be applied correctly. 

```
raw_filtered_chop_clean = apply_ica_replace_mean_std(raw_filtered_chop, ica, picks=picks,
                                                     reject=reject, exclude=ica.exclude,
                                                     n_pca_components=None)

raw_chop_clean = apply_ica_replace_mean_std(raw_chop, ica, picks=picks, reject=reject,
                                            exclude=ica.exclude, n_pca_components=None)

fnout_fig = '109925_CAU01A_100715_0842_2_c,rfDC,0-45hz,ar-perf'
plot_performance_artifact_rejection(raw_chop.copy(), ica, fnout_fig, meg_clean=raw_chop_clean,
                                    name_ecg='ECG 001', name_eog='EOG 002')
```

So, I think we need to double check the function if it is loading the raw correctly.

**Comment by ckiefer0** [comment]
I just tested it again and I got the same results. The components identified were 0 and 1 in both cases.

Setting the random seed to 42 and the iterations to 5000 resulted in MLICA identifying components 0, 1, and 18. Correlation & ctps remained at 0 and 1. The plot still looks good (see below).
![109925_cau01a_100715_0842_2_c rfdc 0-45hz ar-perf_see42 iter5000](https://user-images.githubusercontent.com/25769508/41476537-1d43df3e-70c2-11e8-964e-c87857489d26.png)
Next week I will test what happens using the most recent jumeg version.

**Comment by ckiefer0** [comment]
It seems to be a problem with mne_v0.14.

Using the mne v0.15 and the latest jumeg version, I get the same results as before. However, using mne v0.14 and the latest jumeg version, I get the same result as you.

![109925_cau01a_100715_0842_2_c rfdc 0-45hz ar-perfseed42 iter5000 latestjumeg mne014](https://user-images.githubusercontent.com/25769508/41524806-f654aa56-72de-11e8-9e2c-52be9b6ad7bd.png)

I also tested it without the high-pass filter in the noise reducer but that did not really change anything
![109925_cau01a_100715_0842_2_c rfdc 0-45hz ar-perfseed42 iter5000 latestjumeg mne014 no_hp](https://user-images.githubusercontent.com/25769508/41525192-6f1660dc-72e0-11e8-8bf2-a27fd253b6c7.png)

**Comment by pravsripad** [comment]
@ckiefer0 it could possibly be a filtering issue with the older version of MNE. But since this new routine is part of the new 0.15 version we can keep the example. 

Are you still planning to make changes to return both ica object and cleaned raw file after apply_ica_replace_mean_std? I'd say we simply update the ica object, set the correct std and return it and the user will apply it himself. So, your routine could be something like - ica_update_mean_std. What do you think?

**Comment by ckiefer0** [comment]
I added it, see pull request.

**Comment by pravsripad** [comment]
Closing issue solved in #152
